core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -247,68 +290,99 @@ ByteBuffer readMaxKey() {
-            // *2 in order to get read of the signed vs unsigned limits
-            assert value < Short.MAX_VALUE*2;
-            // set two low bytes of the handle index int
-            entries[item + OFFSET_KEY_LENGTH] =
-                (entries[item + OFFSET_KEY_LENGTH]) | (value & KEY_LENGTH_MASK);
-            return;
-        case OFFSET_KEY_BLOCK:
-            // OFFSET_KEY_LENGTH and OFFSET_KEY_BLOCK should be less then 16 bits long
-            // *2 in order to get read of the signed vs unsigned limits
-            assert value < Short.MAX_VALUE*2;
-            // offset must be OFFSET_KEY_BLOCK,
-            // set 2 high bytes of the int inside OFFSET_KEY_LENGTH
-            assert value > 0; // block id can never be 0
-            entries[item + OFFSET_KEY_LENGTH] =
-                (entries[item + OFFSET_KEY_LENGTH]) | (value << KEY_BLOCK_SHIFT);
-            return;
-        default:
-            entries[item + offset] = value;
-            return;
+            case KEY_LENGTH:
+                // OFFSET_KEY_LENGTH and OFFSET_KEY_BLOCK should be less then 16 bits long
+                // *2 in order to get read of the signed vs unsigned limits
+                assert value < Short.MAX_VALUE * 2;
+                // set two low bytes of the handle index int
+                entries[item + offset.value] =
+                        (entries[item + offset.value]) | (value & KEY_LENGTH_MASK);
+                return;
+            case KEY_BLOCK:
+                // OFFSET_KEY_LENGTH and OFFSET_KEY_BLOCK should be less then 16 bits long
+                // *2 in order to get read of the signed vs unsigned limits
+                assert value < Short.MAX_VALUE * 2;
+                // offset must be OFFSET_KEY_BLOCK,
+                // set 2 high bytes of the int inside OFFSET_KEY_LENGTH
+                assert value > 0; // block id can never be 0
+                entries[item + offset.value] =
+                        (entries[item + offset.value]) | (value << KEY_BLOCK_SHIFT);
+                return;
+            case VALUE_LENGTH:
+                entries[item + offset.value] =
+                        (entries[item + offset.value]) | (value & VALUE_LENGTH_MASK);
+                return;
+            case VALUE_BLOCK:
+                assert value > 0; // block id can never be 0
+                entries[item + offset.value] =
+                        (entries[item + offset.value]) | (value << VALUE_BLOCK_SHIFT);
+                return;
+            default:
+                entries[item + offset.value] = value;
         }
     }
 
     /**
      * gets the value for the given item, or 'null' if it doesn't exist
      */
-    Handle getHandle(int entryIndex) {
-
-        int hi = getEntryField(entryIndex, OFFSET_HANDLE_INDEX);
+    Slice getValueSlice(int entryIndex) {
 
+        long valueReference = getValueReference(entryIndex);
+        int[] valueArray = UnsafeUtils.longToInts(valueReference);
```

Comment:
Why valueArray is needed? Can't getValueReference signal anyhow that the value is not there? Like some invalid long?

Reply:
The INVALID_BLOCK_ID is also checked later inside buildValueSlice()...

Reply:
Will be fixed!